### PR TITLE
Robot Maze updates for Swift 3

### PR DESCRIPTION
**Pull request by EdenShapiro**
Hey @GabrielleM and @saraswatayu, could you guys check and approve these? Thanks

### CODE REVIEW

**Comment by saraswatayu** [comment]
Can we squash these commits into just one?

File: Maze/MazeController.swift [hunk 1]
```diff
@@ -67,8 +67,9 @@ class MazeController {
     }
     
     // MARK: Adding Maze Objects
-    
-    func addMazeObject(var object: MazeObject) {
+    func addMazeObject(obj: MazeObject) {
+        
+        var object = obj
```
Comment (saraswatayu) [inline Maze/MazeController.swift:72]: Not a fan of rewriting it like this; should we consider refactoring the way the Maze works? Shouldn't the MazeView be responsible for adding a maze object (similar to how UIViewController/UIView work)?

File: Maze/MazeController.swift [hunk 2]
```diff
@@ -119,8 +120,10 @@ class MazeController {
         }
     }
     
-    func moveObjectToLocation(var object: MazeObject, newLocation: MazeLocation, completionHandler: () -> Void) {
+    func moveObjectToLocation(obj: MazeObject, newLocation: MazeLocation, completionHandler: () -> Void) {
 
+        var object = obj
```
Comment (saraswatayu) [inline Maze/MazeController.swift:125]: ^ see above. Shouldn't this method belong to the MazeObject? Something like MazeObject.moveToLocation(newLocation: MazeLocation, completionHandler: () -> Void)?

**Comment by EdenShapiro** [comment]
@GabrielleM thoughts?

**Pull request merged**